### PR TITLE
Code Snippets for Java

### PR DESCRIPTION
**Pull request by ramsessanchez**
-Fixes missing schema information in additionalData types. 
-Replaces deltaResponse to deltaGetResponse for java snippets as that is what is generated in Kiota
-Removes the value identifier for java and generated .content() for snippets instead. 
-Handle case where path parameter is an int and not a string, this is done in java and dotnet. 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/1798)

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/OpenAPISnippetsGenerator.cs [hunk 1]
```diff
@@ -96,6 +97,7 @@ public string ProcessPayloadRequest(HttpRequestMessage requestPayload, string la
                 "php" => new PhpGenerator(),
                 "python" => new PythonGenerator(),
                 "cli" => new GraphCliGenerator(),
+                "java" => new JavaGenerator(),
```
Comment (zengin) [inline CodeSnippetsReflection.OpenAPI/OpenAPISnippetsGenerator.cs:100]: Is this going to be deployed in sync with SDK GA?

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/JavaGenerator.cs [hunk 2]
```diff
@@ -229,7 +230,15 @@ private static string GetFluentApiPath(IEnumerable<OpenApiUrlTreeNode> nodes, Sn
                                                                                           .Where(static s => !string.IsNullOrEmpty(s))
                                                                                           .Select(static s => s.ToFirstCharacterUpperCase())
                                                                                           .Aggregate(static (a, b) => $"By{a}{b}");
-                    return $"{pathName ?? "ByTypeId"}{x.Segment.Replace("{", "(\"{", StringComparison.OrdinalIgnoreCase).Replace("}", "}\")", StringComparison.OrdinalIgnoreCase)}.";
+
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/JavaGenerator.cs:233]: Any chance we can try rebase this branch? I didn't expect to see this in the diff after the merge of https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/1773/files

File: CodeSnippetsReflection.OpenAPI/OpenAPISnippetsGenerator.cs (hunk 1 above)
Comment (ramsessanchez) [reply CodeSnippetsReflection.OpenAPI/OpenAPISnippetsGenerator.cs:100]: Ah yes, good catch. Lets hold off on this for a couple more days. Planning RC at the moment and may be a good idea to get all things settled before.

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/JavaGenerator.cs (hunk 2 above)
Comment (ramsessanchez) [reply CodeSnippetsReflection.OpenAPI/LanguageGenerators/JavaGenerator.cs:233]: Will do

**Review by zengin — APPROVED** [review]

**Pull request merged**